homeassistant/components/tellduslive.py

```diff
@@ -5,213 +5,311 @@
+MIN_UPDATE_INTERVAL = timedelta(seconds=5)
+DEFAULT_UPDATE_INTERVAL = timedelta(minutes=1)
 
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Required(CONF_PUBLIC_KEY): cv.string,
         vol.Required(CONF_PRIVATE_KEY): cv.string,
         vol.Required(CONF_TOKEN): cv.string,
         vol.Required(CONF_TOKEN_SECRET): cv.string,
+        vol.Optional(CONF_UPDATE_INTERVAL, default=DEFAULT_UPDATE_INTERVAL): (
+            vol.All(cv.time_period, vol.Clamp(min=MIN_UPDATE_INTERVAL)))
     }),
 }, extra=vol.ALLOW_EXTRA)
 
+API_URL = 'https://api.telldus.com/json/'
+
+# tellstick constants
+TELLSTICK_TURNON = 1
+TELLSTICK_TURNOFF = 2
+TELLSTICK_BELL = 4
+TELLSTICK_TOGGLE = 8
+TELLSTICK_DIM = 16
+TELLSTICK_LEARN = 32
+TELLSTICK_EXECUTE = 64
+TELLSTICK_UP = 128
+TELLSTICK_DOWN = 256
+TELLSTICK_STOP = 512
+
+TELLDUSLIVE_METHODS = {
+    TELLSTICK_TURNON: 'device/turnOn',
+    TELLSTICK_TURNOFF: 'device/turnOff',
+    TELLSTICK_DIM: 'device/dim',
+    TELLSTICK_UP: 'device/up',
+    TELLSTICK_DOWN: 'device/down',
+    TELLSTICK_STOP: 'device/stop',
+}
+
+ATTR_LAST_UPDATED = 'time_last_updated'
+
 
 def setup(hass, config):
     """Setup the Telldus Live component."""
     # fixme: aquire app key and provide authentication using username+password
 
-    global NETWORK
-    NETWORK = TelldusLiveData(hass, config)
+    client = TelldusLiveClient(hass, config)
 
-    if not NETWORK.validate_session():
+    if not client.validate_session():
         _LOGGER.error(
-            "Authentication Error: "
-            "Please make sure you have configured your keys "
-            "that can be aquired from https://api.telldus.com/keys/index")
+            'Authentication Error: '
+            'Please make sure you have configured your keys '
+            'that can be aquired from https://api.telldus.com/keys/index')
         return False
 
-    NETWORK.discover()
+    hass.data[DOMAIN] = client
+    client.update(utcnow())
 
     return True
 
 
-@Throttle(MIN_TIME_BETWEEN_SWITCH_UPDATES)
-def request_switches():
-    """Make request to online service."""
-    _LOGGER.debug("Updating switches from Telldus Live")
-    switches = NETWORK.request('devices/list')
-    # Filter out any group of switches.
-    if switches and 'device' in switches:
-        return {switch["id"]: switch for switch in switches['device']
-                if switch["type"] == "device"}
-    return None
-
-
-@Throttle(MIN_TIME_BETWEEN_SENSOR_UPDATES)
-def request_sensors():
-    """Make request to online service."""
-    _LOGGER.debug("Updating sensors from Telldus Live")
-    units = NETWORK.request('sensors/list')
-    # One unit can contain many sensors.
-    if units and 'sensor' in units:
-        return {(unit['id'], sensor['name'], sensor['scale']):
-                dict(unit, data=sensor)
-                for unit in units['sensor']
-                for sensor in unit['data']}
-    return None
-
-
-class TelldusLiveData(object):
+class TelldusLiveClient(object):
```

Comment:
All this protocol and request related code should be moved to an external lib. It has too many specific things . 

Why did you move away from using tellive-py?